students/robert/Session 7/html_render.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python 
+
+#import io
+
+class Element(object):
+#    tag_indent = {'html':0,'head':4,'meta':8,'title':8,'body':4,'h2':8,'p':8,'hr':8,'ul':8,'li':12}
+    tag = 'html'
+    indent = '    '
+#    content =[]
+
+    def __init__ (self,content=None,**attributes):
+        self.cont = []
+        self.attributes = attributes
+        if content is not None:
+            self.cont.append(content)
+
```

Comment:
It always helps me to first write out what I need to do in pseudo code, and then go through and fill out the code step by step. This way, your code is already commented, and it helps you walk through the logic and the order in which you need to do things. 

So, for part 1, something like: 

# create element class

    # class attribute for html tag
    # class attribute for first level of indentation
    
    # way to store content

    # append method

    # render (file_out, ind = "') method
  
And then fill in the code!